### ISSUE TITLE
Terragrunt manager to support gitlab repos as a module source

### ISSUE THREAD

**Issue by JoshiiSinfield** [renovatebot/renovate#18444]
### What would you like Renovate to be able to do?

We use renovate within gitlab-ci on private runners, within a private gitlab group.

We have an extensive list of terraform modules that sit within private gitlab repos which use semver tags for versioning. We also have a large Terragrunt codebase using many of these modules sourced from gitlab.

We'd like the terragrunt manager to support gitlab-tags for the terraform module source in order to upgrade these module sources automatically.

### If you have any ideas on how this should be implemented, please tell us here.

I think this list of datasources needs gitlab-tags adding to it? 
https://github.com/renovatebot/renovate/blob/5bdbadf24636b5afb1b7d724d5a1d1a79cc93c68/lib/modules/manager/terragrunt/index.ts#L11

### Is this a feature you are interested in implementing yourself?

Yes

**Comment by rarkins** [comment]
It needs more than just adding to that list

**Comment by JoshiiSinfield** [comment]
Hi @rarkins Thanks. I've realised that after doing some more digging into the code.

I've looked at cocoapods which uses the GitlabTagsDatasource. It does the regex check to determine the platform in use & then sets up the gitlabtags as the datasource on the dependency. So we'd need that here in modules.ts - analyzeTerragruntModule

From looking through the code that's all I can see that would need updating? Unless you have any more info? :) 

I'm happy to take a stab at this soon.

**Comment by rarkins** [comment]
Yes, you're on the right track. I was going to suggest find some other managers which use the same datasource and see how they do it. Once you've figured that out, it shouldn't be _too_ complex. Just try to cover both gitlab.com and self-hosted GitLab

**Comment by rarkins** [comment]
BTW see if you can find a terragrunt module hosted on gitlab.com which you can use in a reproduction

**Comment by JoshiiSinfield** [comment]
Hi. I've done the first part of the work, went to push up and saw the CLA. I'm going to have to go through my employer for this to get it authorised. So I'll be delayed with this.

**Comment by patsevanton** [comment]
A very necessary feature!
